torch_geometric/transforms/rooted_subgraphs.py

```diff
@@ -0,0 +1,197 @@
+import torch
+from torch_sparse import SparseTensor
+
+from torch_geometric.data import Data
+from torch_geometric.transforms import BaseTransform
+
+
+class RootedSubgraphData(Data):
+    r"""A data object describing a homogeneous graph together with each node's
+    rooted subgraph. It contains several additional properties that hold the
+    information to map to batch of every node's rooted subgraph.
+
+    Additional Properties:
+        n_id (Tensor): index of nodes includes in combined rooted subgraph.
+        e_id (Tensor): index of edges included in rcombined ooted subgraph.
+        n_sub_batch (Tensor): batch index for nodes in rooted subgraph index.
+        e_sub_batch (Tensor): batch index for edges in rooted subgraph index.
+    """
+    def __inc__(self, key, value, *args, **kwargs):
+        if key in ('n_sub_batch', 'e_sub_batch'):
+            return 1 + getattr(self, 'n_sub_batch')[-1]
+        elif key == 'n_id':
+            return self.num_nodes
+        elif key == 'e_id':
+            return self.edge_index.size(-1)
+        else:
+            return super().__inc__(key, value, *args, **kwargs)
+
+    def map_edge_index(self) -> torch.Tensor:
+        """
+        Map edge_index to the subgraph combined edge_index
+        """
+        edge_index = self.edge_index[:, self.e_id]
+        n_nodes = self.x.size(0)
+        n_sub_nodes = self.n_id.size(0)
+
+        node_map = torch.ones((n_nodes, n_nodes)).long() * -1
```

Comment:
That's a pretty expensive op, but it is indeed tricky to compute. How about we again pre-compute this?

Reply:
Done (required a little bit of refactoring, not much).

Actually, I feel this could be done as a sparse op probably.. but I can leave that for the next PR, there are a couple of improvements to make.